pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -181,6 +181,11 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 		return errors.Errorf("failed to reconcile LB attachment: %+v", err)
 	}
 
+	if machine.Spec.ProviderID == nil || *machine.Spec.ProviderID == "" {
+		providerID := fmt.Sprintf("aws:////%s", *scope.MachineStatus.InstanceID)
```

Comment:
Shouldn't we always reconcile the providerID if it's not there? This is to account for the case that the ProviderID didn't get stored for one reason or another, but we retrieved the instance using tags instead.